chain/chain/src/chain.rs

```diff
@@ -1107,6 +1107,9 @@ impl Chain {
                     byzantine_assert!(false);
                     return Err(Error::InvalidChunk);
                 }
+                if chunk_header.shard_id() != shard_id as ShardId {
```

Comment:
Consider `ShardId::from(shard_id)`. Or `ShardId::try_from(shard_id)` with appropriate handling.

Reply:
Will fix it in a follow up PR. I need to add tests for this PR later. We needed to merge it asap for the release.